BridgeCareApp/BridgeCare/DataAccessLayer/AttributesByYear.cs

```diff
@@ -0,0 +1,112 @@
+﻿using BridgeCare.ApplicationLog;
+using BridgeCare.Interfaces;
+using BridgeCare.Models;
+using BridgeCare.Utility;
+using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Data.SqlClient;
+using System.Linq;
+
+namespace BridgeCare.DataAccessLayer
+{
+    public class AttributesByYear : IAttributesByYear
+    {
+        private readonly BridgeCareContext db;
+
+        public AttributesByYear(BridgeCareContext context)
+        {
+            db = context ?? throw new ArgumentNullException(nameof(context));
+        }
+
+        public List<AttributeByYearModel> GetProjectedAttributes(SimulatedSegmentAddressModel segmentAddressModel, BridgeCareContext db)
+        {
+            if (segmentAddressModel == null)
+                return new List<AttributeByYearModel>();
+
+            var selectStatement = String.Format("SELECT * FROM SIMULATION_{0}_{1} WHERE SectionID={2}",
+                segmentAddressModel.NetworkId, segmentAddressModel.SimulationId, segmentAddressModel.SectionId);
+
+            return GeneralAttributeValueQuery(selectStatement);
+        }
+
+        public List<AttributeByYearModel> GetHistoricalAttributes(SectionModel sectionModel, BridgeCareContext db)
+        {
+            if (sectionModel == null)
+                return new List<AttributeByYearModel>();
+
+            var selectStatement = String.Format(
+                "SELECT * FROM SEGMENT_{0}_NS0 WHERE SectionID={1}",
+                sectionModel.NetworkId, sectionModel.SectionId);
+
+            return GeneralAttributeValueQuery(selectStatement);
+        }
+
+        public List<AttributeByYearModel> GeneralAttributeValueQuery(string selectStatement)
+        {
+            try
+            {
+                DataTable queryReturnValues = UtilityFunctions.NonEntitySQLQuery(selectStatement, db);
+                return DataTableToAttributeList(queryReturnValues);
+            }
+            catch (SqlException ex)
+            {
+                HandleException.SqlError(ex, "Query:" + selectStatement);
+            }
+            catch (OutOfMemoryException ex)
+            {
+                HandleException.OutOfMemoryError(ex);
+            }
+
+            return new List<AttributeByYearModel>();
+        }
+
+        private bool IsNumber(string str) => int.TryParse(str, out _);
+
+        //check for year
+        private bool IsIamYear(string inputString) => (IsNumber(inputString) && inputString.Length == 4);
+
+        private bool IsAttributeYearNameProfile(string columnName, DataTable AttributesDataTable)
+        {
+            string value = AttributesDataTable.Rows[0][columnName].ToString();
+
+            string[] tokens = columnName.Split('_');
+            return (IsIamYear(tokens.Last()) && value.Length > 0);
+        }
+
+        public List<AttributeByYearModel> DataTableToAttributeList(DataTable AttributesDataTable)
```

Comment:
What happens if someone passes a data table that is NOT the ATTRIBUTES_ data table?

Reply:
The routine only pays attention to columns *_YYYY, if there are none then it returns an empty array. Even though, moved it locally and out of the 'adapter class'  to this class to protect against trying to use it in a frivous way